portal/server/types/db-models/users.ts

```diff
@@ -0,0 +1,28 @@
+import { WithId } from 'mongodb';
+import { PortalRole } from '../portal/portal-role';
+import { Bank } from './banks';
+import { UnixTimestamp } from '../date';
+import { Prettify } from '../types-helper';
+
+export type PortalUser = Prettify<
+  WithId<{
+    'user-status': string;
+    disabled?: boolean;
+    timezone: string;
+    firstname: string;
+    surname: string;
+    roles: PortalRole[];
+    email: string;
+    bank: Bank;
+    username: string;
+    salt: string;
+    hash: string;
+    resetPwdToken?: string;
+    resetPwdTimestamp?: string;
+    lastLogin?: UnixTimestamp;
```

Comment:
The type of `PortalSessionUser` should be based on the result of this `portal-api` function:
```
const sanitizeUser = (user) => ({
  username: user.username,
  firstname: user.firstname,
  surname: user.surname,
  email: user.email,
  roles: user.roles,
  bank: user.bank,
  timezone: user.timezone,
  lastLogin: user.lastLogin,
  'user-status': user['user-status'],
  disabled: user.disabled,
  signInLinkSendDate: user.signInLinkSendDate,
  signInLinkSendCount: user.signInLinkSendCount,
  _id: user._id,
});
```

Reply:
We have implemented this type in [this PR](https://github.com/UK-Export-Finance/dtfs2/pull/2751/files#diff-d21b8bbac6f76595a45c1579d2b85699409ed7f03ec706d96b3e765e4491dada) :)

Reply:
fwiw as well, the 2fa process doesnt actually store a `user` (good job past me, I hope)
```
    // We do not store this in the user object to avoid existing logic using the existence of a `user` object to draw elements
    req.session.userEmail = userEmail;
```